cpuprofilenode.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2019 Roger Chapman and the v8go contributors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package v8go
+
+// #include <stdlib.h>
+// #include "v8go.h"
+import "C"
+
+// CPUProfileNode represents a node in a call graph.
+type CPUProfileNode struct {
+	ptr C.CpuProfileNodePtr
+	iso *Isolate
+}
+
+// Returns function name (empty string for anonymous functions.)
+func (c *CPUProfileNode) GetFunctionName() string {
+	str := C.CpuProfileNodeGetFunctionName(c.ptr)
+	return C.GoString(str)
+}
+
+// Retrieves number of children.
+func (c *CPUProfileNode) GetChildrenCount() int {
+	i := C.CpuProfileNodeGetChildrenCount(c.ptr)
+	return int(i)
+}
+
+// Retrieves a child node by index.
+func (c *CPUProfileNode) GetChild(index int) *CPUProfileNode {
+	count := c.GetChildrenCount()
+	if index < 0 || index > count {
+		return nil
+	}
+	ptr := C.CpuProfileNodeGetChild(c.ptr, C.int(index))
+	return &CPUProfileNode{ptr: ptr, iso: c.iso}
+}
+
+// Retrieves the ancestor node, or null if the root.
+func (c *CPUProfileNode) GetParent() *CPUProfileNode {
+	ptr := C.CpuProfileNodeGetParent(c.ptr)
+	return &CPUProfileNode{ptr: ptr, iso: c.iso}
+}
+
+// Returns the number, 1-based, of the line where the function originates.
+func (c *CPUProfileNode) GetLineNumber() int {
+	i := C.CpuProfileNodeGetLineNumber(c.ptr)
```

Comment:
I think supporting the basic APIs from V8 for these components would be a good first step to introducing them into the v8go codebase and seeing how they are used by others and ourselves, as well as what optimizations or additional functionality we could provide. I agree that being able to read the entire cpu profile onto an object once profiling is complete would make it easier for a user to retrieve everything off a single go object instead of making multiple calls and incurring the additional Go<->C costs. This however could easily be a separate function or improvement on `StopProfiling` where it loads everything useful onto a Go CpuProfile object.

Reply:
If there isn't a use case for reading a partial CPU profile, then perhaps we could just change the implementation internally to read everything in C++ upfront, rather than exposing a separate Go API to the user.  Given that you are exposing data through getter functions, the approach you have should allow for this evolution of the implementation without breaking changes.